### PR TITLE
Setup Fly.io

### PR DESCRIPTION
**Pull request by codestronger**
Also posted a message on Teams that gives some context re: this PR, so probably helpful to check that out before diving into this code.

### CODE REVIEW

**Review by nonprofittechy — COMMENTED** [review]
just starting the review, but I really appreciate the detailed operations manual, decision log, etc. this is great, and thank you

File: docs/adr/011-use-papertrail-to-increase-access-to-logs.md
Comment (BryceStevenWilley) [inline docs/adr/011-use-papertrail-to-increase-access-to-logs.md:12]: This all seems great. @nonprofittechy , you might want to make sure that a docassemble integration with papertrail wouldn't be too hard, so eventually all of the logs could reside in the same place.

File: docs/adr/010-use-a-platform-and-database-as-services-to-improve-operations.md
Comment (BryceStevenWilley) [inline docs/adr/010-use-a-platform-and-database-as-services-to-improve-operations.md:40]: I didn't see you change any specific Database configuration in this PR: does that mean that all you need for supabase is the DB Hostname / port? If that's the case, I'd expect migrating to other DBaaSes would be just as easy.

Might be worth noting that it seems like a fairly easy decision to change if necessary.

**Review by BryceStevenWilley — COMMENTED** [review]
Review in progress; I've read everything that's not the operational model. I'll get to that soon.

Everything else looks good so far though!

File: pom.xml
Comment (codestronger) [reply pom.xml:266]: Ah, Janino was needed for the conditional Papertrail appender in the logback.xml. We used that to avoid having to write the same logic as part of a custom appender.

File: src/main/java/edu/suffolk/litlab/efspserver/ecf4/TylerModuleSetup.java
Comment (codestronger) [reply src/main/java/edu/suffolk/litlab/efspserver/ecf4/TylerModuleSetup.java:234]: Oh, that's a good idea! I'll make the change to allow both schedules.

Two reasons why it turned off the scheduled update, neither of which is that strong:

1. The original commented out code alternated between the two options, so I thought they were mutually exclusive use cases
2. At the time, my intention was to have the code updater in a daily scheduled machine run. That is a short-lived machine instance that is shut down as soon as the code updater finishes. However, that Fly.io feature didn't feel that mature yet so we decided not to rely on it. The operational model was to keep 1 hot instance to avoid the cold start delay, which meant there was no reason not to keep the existing Quartz scheduled update.

File: src/main/java/edu/suffolk/litlab/efspserver/ecf4/TylerModuleSetup.java
Comment (codestronger) [reply src/main/java/edu/suffolk/litlab/efspserver/ecf4/TylerModuleSetup.java:204]: Sounds good! I'll update along w/ the documentation.

Looking at this again, I don't think it's worth it to even execute the rest of the logic if the schedule is disabled, since there isn't a mechanism to re-enable the schedule without a restart. Will probably change this to short-circuit return from the method call.

File: docs/adr/010-use-a-platform-and-database-as-services-to-improve-operations.md
Comment (codestronger) [reply docs/adr/010-use-a-platform-and-database-as-services-to-improve-operations.md:40]: Sounds good! I'll make a note that this is an easy decision to change in the future. Any Postgres DBaaS should be easily swappable.

File: fly.yml.example
Comment (codestronger) [reply fly.yml.example:1]: Ah, yes, we can create a real version of the fly.yml under .github/workflows once we decide on how/if we want continuous deployments. We haven't had that conversation yet. I've have to create the environment-specific token so that the workflow won't throw a bunch of errors. I'd lean towards continuous deployment to efsp-staging and leave prod as a manual deployment. Based on the current volume of code changes, keeping only manual deployments is a viable model and leaves this mentally consistent.

File: docs/operations_guide/index.md
Comment (codestronger) [reply docs/operations_guide/index.md:40]: Ah, good to know. Will update to note that. We definitely want to play nice w/ Tyler so it sounds like caching that in a shared place is necessary before any future scale out.

File: docs/operations_guide/index.md
Comment (codestronger) [reply docs/operations_guide/index.md:34]: Noted. Will check with @johnphamvan to see if we'll be setting that up.

File: docs/operations_guide/index.md
Comment (codestronger) [reply docs/operations_guide/index.md:5]: Updated.

**Review by BryceStevenWilley — APPROVED** [review]
Sorry it's taken so long to approve this, but finally got through the ops guide. It all looks very good!

I know we ran into some issues trying it out, those are still in progress in https://github.com/SuffolkLITLab/EfileProxyServer/pull/185. I'll try to fix that so CI can pass before this merges. Is there a last date you can work on this @codestronger? I can make sure I get that in with enough time for you to finish up.

**Comment by codestronger** [comment]
> Sorry it's taken so long to approve this, but finally got through the ops guide. It all looks very good!
> 
> I know we ran into some issues trying it out, the fixes for those issues are still in progress in #185. I'll try to fix that so CI can pass before this merges. Is there a last date you can work on this @codestronger? I can make sure I get that in with enough time for you to finish up.

I think the official wrap is June 21, but I can continue to work with you on it after that date. I'd like take the time to get things right and not rush it. Thanks again for all the help and guidance!

File: fly_create_api_key.sh
Comment (codestronger) [reply fly_create_api_key.sh:1]: Updated.

File: fly_create_prod_api_key.sh
Comment (codestronger) [reply fly_create_prod_api_key.sh:1]: Updated.

File: docs/operations_guide/index.md
Comment (codestronger) [reply docs/operations_guide/index.md:77]: Updated for all command examples.

File: docs/operations_guide/index.md
Comment (codestronger) [reply docs/operations_guide/index.md:126]: Updated.

File: docs/operations_guide/index.md
Comment (codestronger) [reply docs/operations_guide/index.md:128]: Updated.

File: docs/operations_guide/index.md
Comment (codestronger) [reply docs/operations_guide/index.md:135]: Maybe we should rewrite this to tone down the language. It's a bit alarmist upon re-reading it. Fly.io stores the docker images in their own registry, that's scoped to our app. There's a layer of auth there, although it's not on the same level as the typical secrets manager. We had a discussion about this topic a while back and the decision was to err on the side of caution. However, anyone with access to the registry would likely also have access to the machines, so it's not that big a deal to bake in secrets even though that's not a best practice.

**Review by samglover — COMMENTED** [review]
I'll defer to @BryceStevenWilley about whether this is ready to merge.

**Pull request merged**
